LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIStrategyMapItem_Region_LW.uc

```diff
@@ -38,6 +17,48 @@ var localized string m_strMonthlyRegionalIncome;
 var string CachedRegionLabel;
 var string CachedHavenLabel;
 
+simulated function bool CanMakeContact()
+{
+	return (class'UIUtilities_Strategy'.static.GetXComHQ().IsContactResearched() && (GetRegion().ResistanceLevel == eResLevel_Unlocked));
+}
+
+simulated function bool CanMakeRadioRelay()
+{
+	return (class'UIUtilities_Strategy'.static.GetXComHQ().IsOutpostResearched() && (GetRegion().ResistanceLevel == eResLevel_Contact));
+}
+
+simulated function bool IsRadioRelayInstalled()
+{
+	return (GetRegion().ResistanceLevel == eResLevel_Outpost);
+}
+
+/* Issue # 815 : KDM : When using a controller, UIStrategyMap continuously calls UpdateSelection() --> SelectMapItemNearestLocation().
+Now, within SelectMapItemNearestLocation() there is a loop which goes through each XComGameState_GeoscapeEntity and determines if its associated 
+strategy map UI item is potentially selectable. The function IsSelectable() within UIStrategyMapItem_Region is rather odd since :
+1.] It returns false if IsResHQRegion() is true; in other words, the HQ region isn't selectable.
+2.] It returns false if the scan button type isn't eUIScanButtonType_Default, eUIScanButtonType_Contact, or eUIScanButtonType_Tower. 
+	Yet, the default scan button type is EUIScanButtonType_MAX which isn't even part of the enumeration.
+3.] It doesn't appear to notice a region once it has a relay installed (eResLevel_Outpost).
+
+I have determined that :
+1.] Whether a region is the HQ or not should have no bearing on selectability so ignore it entirely.
+2.] The scan button type should have no bearing on selectability so ignore it entirely.*/
+simulated function bool IsSelectable()
+{
+	// KDM : The region is selectable if any of these conditions are true :
+	// 1.] It is contactable, regardless of whether contacting has commenced.
+	// 2.] It has been contacted, and a radio relay can be built, regardless of whether this building process has actually commenced.
+	// 3.] It has a radio relay installed.
+	if (CanMakeContact() || CanMakeRadioRelay() || IsRadioRelayInstalled())
```

Comment:
You can just return the value of this condition (see my earlier comment). `if` branches that *just* return `true` or `false` and do nothing else can generally be compressed into a single `return`.

Reply:
Done.